action.yml

```diff
@@ -90,10 +90,10 @@ runs:
         
         echo DOCKER_REGISTRY="${{ inputs.maven-artifacts-account-id }}.dkr.ecr.${AWS_REGION}.amazonaws.com" >> $GITHUB_ENV
         
-        IS_SELF_HOSTED_RUNNER="false"
-        if [[ "$USER" == "ec2-user" ]];
-          then
-            IS_SELF_HOSTED_RUNNER="true"
+        IS_SELF_HOSTED_RUNNER="true"
+
+        if [[ "${{ runner.name }}" =~ ^GitHub\ Actions.* ]]; then
```

Comment:
based on this hint: https://github.com/orgs/community/discussions/48359 hopefully won't be changed